### ISSUE TITLE
no-undefined-types regression for imported module types in v51.0.0

### ISSUE THREAD

**Issue by Andrioden** [gajus/eslint-plugin-jsdoc#1412]
## Expected behavior
Eslint pass OK without warning

## Actual behavior
Eslint warns `The type 'Ably.Realtime' is undefined   jsdoc/no-undefined-types`.

On v50.8.0 this do not occur.

## ESLint Config

I have a lot of rules, but no related config to this warning

## ESLint sample

External dependency (happens to both in exact same pattern as below)

package.json
``` js
  "dependencies": {
    "ably": "2.9.0",
    "konva": "9.3.20",
  },
```

code.js
``` js
import * as Ably from "ably"

export class Code {
    /** @type {Ably.Realtime} */
    static #client
}
```

## Environment

- Node version: v22.14.0
- ESLint version eslint@9.29.0
- `eslint-plugin-jsdoc` version: v51.0.0

**Comment by azat-io** [comment]
Same problem.

<img width="920" alt="Image" src="https://github.com/user-attachments/assets/ab6ee8f8-c684-43d7-9eaa-f9965f179f75" />
